posthog/batch_exports/service.py

```diff
@@ -847,3 +849,46 @@ async def afetch_batch_export_runs_in_range(
     ).order_by("data_interval_start")
 
     return [run async for run in queryset]
+
+
+def fetch_earliest_backfill_start_at(
+    *,
+    team_id: int,
+    model: str,
+    interval_time_delta: dt.timedelta,
+    exclude_events: collections.abc.Iterable[str] | None = None,
+    include_events: collections.abc.Iterable[str] | None = None,
+) -> dt.datetime:
+    """Get the earliest start_at for a batch export backfill.
+
+    Note that this query will return the unix epoch if no events or persons exist.
+    """
+    interval_seconds = interval_time_delta.total_seconds()
+    if model == "events":
+        exclude_events = exclude_events or []
+        include_events = include_events or []
+        query = """
+            SELECT toStartOfInterval(MIN(timestamp), INTERVAL %(interval_seconds)s SECONDS)
+            FROM events
+            WHERE team_id = %(team_id)s
+            AND (length(%(include_events)s::Array(String)) = 0 OR event IN %(include_events)s::Array(String))
+            AND (length(%(exclude_events)s::Array(String)) = 0 OR event NOT IN %(exclude_events)s::Array(String))
+        """
+        query_args = {
+            "team_id": team_id,
+            "include_events": include_events,
+            "exclude_events": exclude_events,
+            "interval_seconds": interval_seconds,
+        }
+    elif model == "persons":
+        query = """
+            SELECT toStartOfInterval(MIN(_timestamp), INTERVAL %(interval_seconds)s SECONDS)
+            FROM person
+            WHERE team_id = %(team_id)s
+        """
```

Comment:
Why not query `persons_batch_export`? Is it perhaps too expensive?

Reply:
Could do. I decided against it since it requires an `interval_start` and `interval_end` but suppose these could just be set to something very low and high. Maybe I'll do some tests for a few teams to see if there's any difference between my query and the `persons_batch_export` and what the performance difference is

Reply:
Proooooooooooobably fine to just do persons. And there is a perf. cost to hitting two tables.